src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -133,25 +141,34 @@ def export_config(cmd,
                   key=None,
                   prefix="",  # prefix to remove
                   yes=False,
+                  skip_features=False,
                   # to-file parameters
                   path=None,
                   format_=None,
                   separator=None,
+                  naming_convention='pascal',
                   # to-config-store parameters
                   dest_name=None,
                   dest_connection_string=None,
                   dest_label=None,
+                  preserve_labels=False,
                   # to-app-service parameters
-                  appservice_account=None,
-                  skip_features=False,
-                  naming_convention='pascal'):
+                  appservice_account=None):
     src_features = []
     dest_features = []
     dest_kvs = []
     destination = destination.lower()
     format_ = format_.lower() if format_ else None
     naming_convention = naming_convention.lower()
 
+    if destination == 'appconfig' and dest_label is not None and preserve_labels:
+        raise CLIError("Export failed! Please provide only one of these arguments: 'dest-label' or 'preserve-labels'. See 'az appconfig kv export -h' for examples.")
+    if destination == 'appconfig' and preserve_labels:
+        # We need dest_label to be the same as label for preview later.
+        # This will have no effect on label while writing to config store
+        # as we check preserve_labels again before labelling KVs.
+        dest_label = label
```

Comment:
nit: to make the logic better understanding, can we use nested if statement 
```
if destination == 'appconfig':
    if dest_label is not None and preserve_labels
          raise error
    if preserve_labels
         dest_label = label
```

Reply:
Sure

---
In reply to: [363897991](https://github.com/Azure/azure-cli/pull/11756#discussion_r363897991) [](ancestors = 363897991)